packages/schema/parse/src/version_compare/comparators.ts

```diff
@@ -0,0 +1,489 @@
+import {
+  QueryDefinition,
+  EnumDefinition,
+  ObjectDefinition,
+  PropertyDefinition,
+  GenericDefinition,
+  ArrayDefinition,
+  MethodDefinition,
+  ImportedQueryDefinition,
+  ImportedEnumDefinition,
+  ImportedObjectDefinition,
+  ImportedDefinition,
+} from "../typeInfo/definitions";
+import { TypeInfo } from "../typeInfo";
+import { compareSets, SetComparisionType } from "./setUtils";
+
+export type ImportDefinition = { type: string };
+
+export enum VersionRelease {
+  PATCH,
+  MINOR,
+  MAJOR,
+}
+
+export function compare(t1: TypeInfo, t2: TypeInfo): VersionRelease {
```

Comment:
While I find it interesting to be able to know what the "VersionRelease" of two schemas is, there is no use for it currently(as we want to either just check for patch or just check for minor). If in the future there does arise some use for it then we can implement it. Right now, it's just an unnecessary performance drain.

Also, I'd say we are more interested in schema comparison than version comparison, since the former is a subset of the latter. So we need two functions: 
```typescript
export const areTypeInfosFunctionallyIdentical = (
  typeInfoA: TypeInfo,
  typeInfoB: TypeInfo
): boolean
```
and
```typescript
export const areTypeInfosBackwardCompatible = (
  baseTypeInfo: TypeInfo,
  derivedTypeInfo: TypeInfo
): boolean
```

Note the absence of minor and patch being mentioned